### PR TITLE
Fix/build user vars plugin

### PR DESCRIPTION
**Pull request by mvignesh12**
- Fixed Post build query not updating the dashboard with expanded values of variables in additional columns set with build user vars plugin
- Fixed Post-Build Update: WARN: Could not alter table env_dashboard to add column

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java [hunk 1]
```diff
@@ -140,7 +140,15 @@ public void tearDown(Run<?, ?> build, FilePath workspace, Launcher launcher, Tas
 				String passedBuildJob = build.getEnvironment(listener).expand(buildJob);
 				String passedPackageName = build.getEnvironment(listener).expand(packageName);
 				String doDeploy = build.getEnvironment(listener).expand("$UPDATE_ENV_DASH");
-				List<ListItem> passedColumnData = Collections.emptyList();
+				//List<ListItem> passedColumnData = Collections.emptyList();
```
Comment (willwh) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: Please remove this commented line

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java [hunk 2]
```diff
@@ -172,7 +180,7 @@ public void tearDown(Run<?, ?> build, FilePath workspace, Launcher launcher, Tas
 		context.setDisposer(new TearDownImpl());
 	}
 
-	@SuppressWarnings("rawtypes")
```
Comment (willwh) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: why did you remove this suppress?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java [hunk 3]
```diff
@@ -211,12 +219,12 @@ private String writeToDB(Run<?, ?> build, TaskListener listener, String envName,
...
 			contents = contents + "', '" + item.contents;
 			try {
-				stat.execute("ALTER TABLE env_dashboard ADD IF NOT EXISTS " + item.columnName + " VARCHAR;");
+				stat.execute("ALTER TABLE env_dashboard ADD COLUMN IF NOT EXISTS " + item.columnName + " VARCHAR;");
 			} catch (SQLException e) {
 				returnComment = "WARN: Could not alter table env_dashboard to add column " + item.columnName + ".";
 				return returnComment;
 			}
-		}
+		}
```
Comment (willwh) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: extraneous whitespace

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java [hunk 4]
```diff
@@ -236,19 +244,24 @@ private String writeToDB(Run<?, ?> build, TaskListener listener, String envName,
...
-		}
-
+		}		
 		String runQuery = null;
 		if (runTime.equals("PRE")) {
 			runQuery = "INSERT INTO env_dashboard (envComp, jobUrl, buildNum, buildStatus, envName, compName, created_at, buildJobUrl, packageName"
 					+ columns + ") VALUES( '" + indexValueofTable + "', '" + currentBuildUrl + "', '" + currentBuildNum
 					+ "', '" + currentBuildResult + "', '" + envName + "', '" + compName + "' , + current_timestamp, '"
-					+ buildJobUrl + "' , '" + packageName + contents + "');";
+					+ buildJobUrl + "' , '" + packageName + contents + "');";
```
Comment (willwh) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: erroneous whitespace

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java [hunk 5]
```diff
@@ -260,7 +273,7 @@ private String writeToDB(Run<?, ?> build, TaskListener listener, String envName,
 			stat.execute(runQuery);
 		} catch (SQLException e) {
 			returnComment = "Error running query " + runQuery + ".";
-			return returnComment;
+			return returnComment + e;
```
Comment (willwh) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: This will fail, `e` is not a string?

**Comment by willwh** [comment]
Thanks very much for having a crack at this @mvignesh12 - just some nitpicks to get tidied up and I think we can merge this!

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java (hunk 1 above)
Comment (mvignesh12) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: Comment removed

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java (hunk 2 above)
Comment (mvignesh12) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: I've added it back. I removed it during code modifications.

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java (hunk 3 above)
Comment (mvignesh12) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: Removed whitespace

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java (hunk 4 above)
Comment (mvignesh12) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: Removed whitespace

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java (hunk 5 above)
Comment (mvignesh12) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/DashboardBuilder.java]: Added it to print the exact exception which I was getting during debugging. Removed it now.

**Comment by mvignesh12** [comment]
Thanks Will. I've updated the code based on your reviews. Please help in merging this PR and release the latest version.

**Comment by willwh** [comment]
@mvignesh12 great, thank you! I'd like to do a little testing this weekend, but I think this is looking good for a merge here. To be able to do another release of the plugin, we'd like to get this migrated over to the `jenkinsci` github team, which is blocking us releasing pipeline support. Hopefully @vipinsthename can assist in getting this migrated over soon :)
